src/components/text.js

```diff
@@ -0,0 +1,372 @@
+    var promise = loadedFontPromises[font] = loadedFontPromises[font] || loadBMFontPromise(font);
+    promise.then(function (loadedFont) {
+      if (loadedFont.pages.length !== 1) {
+        throw new Error('Currently only single-page bitmap fonts are supported.');
+      }
+
+      // Update geometry given font metrics.
+      var data = coerceData(self.data);
+      var textRenderWidth = data.wrappixels || (data.wrapcount * 0.6035 * loadedFont.info.size);
+      var options = assign({}, data, { font: loadedFont, width: textRenderWidth, lineHeight: data.lineHeight || loadedFont.common.lineHeight });
+      var object3D;
+      geometry.update(options);
+      self.mesh.geometry = geometry;
+
+      // Add mesh if not already there.
+      object3D = self.el.object3D;
+      if (object3D.children.indexOf(self.mesh) === -1) {
+        object3D.add(self.mesh);
+      }
+
+      // Look up font image URL to use, and perform cached load.
+      var src = self.data.fontImage || font.replace('.fnt', '.png') || path.dirname(data.font) + '/' + loadedFont.pages[0];
+      var texpromise = loadedTexturePromises[src] || loadTexturePromise(src);
+      texpromise.then(function (image) {
+        // Make mesh visible and apply font image as texture.
+        self.mesh.visible = true;
+        if (image) {
+          self.texture.image = image;
+          self.texture.needsUpdate = true;
+        }
+      }).catch(function () {
+        console.error('Could not load bmfont texture "' + src +
+          '"\nMake sure it is correctly defined in the bitmap .fnt file.');
+      });
+
+      self.currentFont = loadedFont;
+      self.updateLayout(data);
+    }).catch(function (error) {
+      throw new Error('Error loading font ' + self.data.font +
+          '\nMake sure the path is correct and that it points' +
+          ' to a valid BMFont file (xml, json, fnt).\n' + error.message);
+    });
+  },
+
+  updateLayout: function (data) {
+    var el = this.el;
+    var font = this.currentFont;
+    var geometry = this.geometry;
+    var layout = geometry.layout;
+    var elGeo = el.getAttribute('geometry');
+    var width;
+    var textRenderWidth;
+    var textScale;
+    var height;
+    var x;
+    var y;
+    var anchor;
+    var baseline;
+
+    // Determine width to use.
+    width = data.width || (elGeo && elGeo.width) || DEFAULT_WIDTH;
+    // Determine wrap pixel count, either as specified or by experimentally determined fudge factor.
+    // (Note that experimentally determined factor will never be correct for variable width fonts.)
+    textRenderWidth = data.wrappixels || (data.wrapcount * 0.6035 * font.info.size);
+    textScale = width / textRenderWidth;
+    // Determine height to use.
+    height = textScale * geometry.layout.height;
+
+    // update geometry dimensions to match layout, if not specified
+    if (elGeo) {
+      if (!elGeo.width) { el.setAttribute('geometry', 'width', width); }
```

Comment:
If you have a circle geometry, it won't have a `width` value but this code will set that value and it won't affect that geometry. 
And also this is assuming that the user always wants to modify the geometry (box, plane) to fit the text, but it can't be always true.